glean_parser/javascript.py

```diff
@@ -37,6 +51,17 @@ def iterencode(self, value):
                     yield from self.iterencode(subvalue)
                     first = False
                 yield "]"
+            elif isinstance(value, metrics.Rate):
```

Comment:
Why does rate require special handling?

Reply:
The first 2 `rate` metrics work fine, but since `rate_external` has `denominator_metric` it cannot serialize that property (https://github.com/mozilla/glean_parser/blob/main/tests/data/all_metrics.yaml#L113).

It was throwing an error - `TypeError: Object of type Rate is not JSON serializable`. The difference is that in most cases it is serializing the properties of all the objects, which are serializable types. Since `rate_external` has a `rate` metric are one of its properties, rather than trying to serialize a string or number, it now has to try and serialize another `rate` object, which it can't by default.

For reference, this block is the same as what we are adding for both `swift` and `kotlin` to make sure we serialize properly
- swift - https://github.com/mozilla/glean_parser/blob/main/glean_parser/swift.py#L74
- kotlin - https://github.com/mozilla/glean_parser/blob/main/glean_parser/kotlin.py#L72